dma.c

```diff
@@ -255,8 +254,10 @@ add_recp(struct queue *queue, const char *str, int expand)
 				endpwent();
 			}
 		}
-	} else {
+	} else if (config.features & DELIVERY_REMOTE) {
```

Comment:
If you made this check down in `deliver()` where `deliver_remote` is called you can use `bounce` on the email which might be better behavior if local delivery is allowed.  Of course if both local and remote delivery are disabled we probably want to catch that at configuration time.